tests/poll.rs

```diff
@@ -201,10 +208,112 @@ fn registry_behind_arc() {
     handle2.join().unwrap();
 }
 
+// Testing on Windows only where this must fail by design, on other platforms
+// behavior is undefined.
+#[test]
+#[cfg(target_os = "Windows")]
+fn register_deregister_multiple_poll_instances() {
+    /*
+    This test checks the following register/deregister contrain:
+    A handle can be passed back to `register` after it has been
+    deregistered; however, it must be passed back to the **same** `Poll`
+    instance.
+    */
+
+    let poll1 = Poll::new().expect("unable to create Poll instance");
+
+    let mut listener = TcpListener::bind(any_local_address()).unwrap();
+
+    poll1
+        .registry()
+        .register(&mut listener, ID1, Interest::READABLE)
+        .expect("unable to register listener");
+
+    poll1
+        .registry()
+        .deregister(&mut listener)
+        .expect("unable to deregister listener");
+
+    let poll2 = Poll::new().expect("unable to create Poll instance");
+
+    assert_eq!(
+        poll2
+            .registry()
+            .register(&mut listener, ID1, Interest::READABLE)
+            .unwrap_err()
+            .kind(),
+        io::ErrorKind::Other
+    );
+
+    poll1
+        .registry()
+        .register(&mut listener, ID1, Interest::READABLE)
+        .expect("unable to register listener");
+
+    poll1
+        .registry()
+        .deregister(&mut listener)
+        .expect("unable to deregister listener");
+}
+
+#[test]
+fn reregister_overrides_interest() {
+    /*
+    This test checks the followin reregister constraints
```

Comment:
typo: "following"

Missing the ":" after "constraints".

Reply:
done, corrected.